packages/headless/src/controllers/history-manager/headless-history-manager.test.ts

```diff
@@ -52,6 +57,17 @@ describe('History Manager', () => {
     expect(engine.actions[1].type).toBe(undo().type);
   });
 
+  it('should allow to navigate backward on no results when there is a past history', () => {
+    engine.state.history.present = getHistoryInitialState();
+    engine.state.history.past = [extractHistory({pipeline: 'test'})];
+    initHistoryManager();
+
+    historyManager.backOnNoResults();
+
+    expect(engine.actions[0].type).toBe(back.pending.type);
+    expect(engine.actions[1].type).toBe(undo().type);
+  });
```

Comment:
Where does the `undo` action come from? I do not see it dispatched [here](https://github.com/coveo/ui-kit/pull/1202/files#diff-8b720ca80c59926649cfac18e286e82eb394d213a27c3bdc8ba6eb0b0a1e1169R95).

Also, should there be an assertion for the `executeSearch` action?

Reply:
`back()` is dispatching two action under the hood: `undo` (move back in history) + `change` (update all slice to the new present state).

I can add a check on `executeSearch` also